server/PokerTableObjects/PokerHandRankCalculator.js

```diff
@@ -0,0 +1,421 @@
+        }
+        this.removeSomeMatchingKeys(numbers, cardNumber, triplet);
+        let remainingCards = this.getRemainingCards(numbers);
+        
+        if(setup.number === 0){ // another function called this and only wants the cards
+            return {tripletResult: tripletResult, remainingCards: remainingCards};
+        }
+        // if we made it to this point then send back the full "3 of a kind" status
+        let highCardsResult = this.handHighCardValue(remainingCards, {number: setup.number, condition: false});
+        if(highCardsResult === -1){
+            return -1; // error
+        }
+        let result = 0x400000 + tripletResult + highCardsResult;
+        return result;
+    },
+
+    handTwoPairValue: function(cards, setup){
+        let numbers = this.getTheNumbersArrayA14(cards); // get the numbers
+        numbers.sort(function(a, b){return a - b}); // sort the numbers
+        let twoToAceCount = this.getTwoToAceCount(numbers); // sort the numbers into groups
+        let cardNumber = this.findHighestGroup(twoToAceCount, 2); // find first pair
+        if(cardNumber === -1){
+            return -1;
+        }
+        let pair = 2;
+        let firstPairResult = cardNumber * this.integerPow(16, setup.number + pair + 1) + cardNumber * this.integerPow(16, setup.number + pair);
+        this.removeSomeMatchingKeys(numbers, cardNumber, pair);
+        let remainingCards = this.getRemainingCards(numbers);
+        let secondPairResult = this.handSinglePairValue(remainingCards, setup); // find second pair of remaining cards
+        if(secondPairResult === -1){
+            return -1;
+        }
+        let result = 0x300000 + firstPairResult + (secondPairResult - 0x200000); // remove the single pair status
+        return result;
+    },
+
+    handSinglePairValue: function(cards, setup){ // find the highest pair
+        let numbers = this.getTheNumbersArrayA14(cards);
+        let pair = 2;
+        numbers.sort(function(a, b){return a - b}); // now the array is sort low to high
+        let twoToAceCount = this.getTwoToAceCount(numbers);
+        let cardNumber = this.findHighestGroup(twoToAceCount, pair);
+        if(cardNumber === -1){
+            return -1;
+        }
+        let pairResult = cardNumber * this.integerPow(16, setup.number + 1) + cardNumber * this.integerPow(16, setup.number);
+        if(setup.number === 0){
+            return pairResult;
+        }
+        this.removeSomeMatchingKeys(numbers, cardNumber, pair);
+        let remainingCards = this.getRemainingCards(numbers);
+        let highCardResult = this.handHighCardValue(remainingCards, {number: setup.number, condition: false}); // if full house is calling this then highCardSendBack = 0, and 0 is returned
+        let result = 0x200000 + pairResult + highCardResult;
+        return result;
+    },
+
+    handHighCardValue: function(cards, setup){ // find the highcards
+        let oldAce = 1;
+        let newAce = 14;
```

Comment:
I'm seeing the exact same constant variables pop up multiple times in the code.  For this one specifically, these variables also appear in getTheNumbersArrayA4 (line 372) and handFlushValue (line 156).  If there are variables like these that are repeated multiple times throughout the file or are common between more than one file, does it make sense to put them in a separate file that more than one component can access?

Reply:
these values are only found in the one file 'PokerHandRankCalculator',  ive updated them to be shared among functions